packages/grid/src/vaadin-grid-column-group.js

```diff
@@ -119,7 +118,7 @@ class GridColumnGroup extends ColumnBaseMixin(PolymerElement) {
    * @protected
    */
   _columnPropChanged(path, value) {
-    if (path === 'hidden') {
+    if (path === 'hidden' && !this._preventHiddenCascade) {
```

Comment:
This can be called multiple times [while new columns are being added](https://github.com/vaadin/web-components/blob/master/packages/grid/src/vaadin-grid-column-group.js#L316-L319), and the `this._preventHiddenCascade = false` below could lead to resetting the flag prematurely.